src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -18,12 +18,41 @@ public ARMEmitter(NodeFactory factory, bool relocsOnly)
         public ObjectDataBuilder Builder;
         public TargetRegisterMap TargetRegister;
 
+        // mov reg, immediate
+        // reg range: 0-7
```

Comment:
> By the way, I'm surprised that no one has done such a code for x64

The x64 emitter does not tend to have limitations on which registers can be used where, so such asserts are not needed in it.